### PR TITLE
eval framework update

### PR DESCRIPTION
**Pull request by bdzimmer**
- [X] refactor after tools refactor
- [X] verify database script
    - [X] use relative paths for ids in database 
- [x] eval questions for tools
    - [x] codebase qa
    - [x] docstring
    - [x] lint file
- [x] agent as a "tool" that you can test with a sequence of user prompts

Note that it's not expected that all evals pass. This is a first iteration of the framework / tests and I think we will be adding a lot over the next week.

### CODE REVIEW

**Review by cameronfabbri — APPROVED** [review]

File: data/evaluation.yaml [hunk 1]
```diff
@@ -1,14 +1,90 @@
...
+        type: contains_all
+        checks: ['is None', 'lambda']
+    - name: lint C
+      tool: LintFileTool
+      params:
+        user_prompt: What are some import issues in agents/tools.py
+        filepath_inexact: agents/tools.py
+      eval_func:
+        type: contains_all
+        checks: ['json', 'sqlite3', 'typing.Any', 'typing.List', 'pandasql']
```
[the remark this answers is not in the document]
Comment (bdzimmer) [reply data/evaluation.yaml:45]: That goes for any of these tests, they are all based on a specific state of the codebase. For now there is a check that ensure the codebase is at a specific commit. We could eventually have this be defined per suite or something and have that code automatically be checked out but that seemed like overkill for now.

File: scripts/dbverify.py
Comment (jmpounders) [inline scripts/dbverify.py:1]: Nice

File: src/groundcrew/utils.py [hunk 2]
```diff
@@ -97,12 +97,15 @@ def build_llm_completion_client(
...
-            response = completion(messages)
-            return response.content
+            try:
+                messages = [
+                    openaiapi.SystemMessage("You are a helpful assistant."),
+                    openaiapi.UserMessage(prompt)
+                ]
+                response = completion(messages)
+                return response.content
+            except:
```
Comment (jmpounders) [inline src/groundcrew/utils.py:107]: Bare exception.  Not a strong opinion in this situation, but `except Exception:` will placate the linter.

**Review by jmpounders — APPROVED** [review]
Nice work!  I like the evaluation yaml approach and extendable suite of test functions.

File: src/groundcrew/utils.py (hunk 2 above)
Comment (bdzimmer) [reply src/groundcrew/utils.py:107]: Good catch!

**Pull request merged**